main/vstring.c

```diff
@@ -81,12 +81,12 @@ extern vString *vStringNew (void)
 
 extern void vStringPut (vString *const string, const int c)
 {
-	if (string->length + 1 == string->size)  /*  check for buffer overflow */
+	if (string->length + 1 >= string->size)  /*  check for buffer overflow */
```

Comment:
After thinking this is wrong approach.
As the desing of vString, the condition, `string->length + 1 == string->size`, must be enough.